app/experimenter/nimbus-ui/src/components/PreviewURL/index.tsx

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from "react";
+import Select from "react-select";
+import { SelectOption } from "../../hooks/useCommonForm/useCommonFormMethods";
+import { getExperiment_experimentBySlug } from "../../types/getExperiment";
+
+type CopyableURLProps = {
+  experimentSlug: string;
+  branch: string;
+  collection?: string;
+};
+
+const CopyableURL: React.FC<CopyableURLProps> = ({
+  experimentSlug,
+  branch,
+  collection,
+}) => {
+  let url = `about:studies?optin_slug=${experimentSlug}&optin_branch=${branch}`;
+  const clipboard = navigator.clipboard;
+  if (collection) {
+    url = `${url}&optin_collection=${collection}`;
+  }
+  return (
+    <code className="click-to-copy" onClick={() => clipboard.writeText(url)}>
+      {url}
+    </code>
+  );
+};
+
+export const PreviewURL: React.FC<
+  Pick<
+    getExperiment_experimentBySlug,
+    "slug" | "referenceBranch" | "treatmentBranches"
+  >
+> = ({ slug: experimentSlug, referenceBranch, treatmentBranches }) => {
+  const [selection, setSelection] = React.useState(referenceBranch?.slug || "");
+  const slugs: SelectOption[] = [];
+
+  if (!referenceBranch || !Array.isArray(treatmentBranches)) {
+    return null;
+  }
+
+  [referenceBranch.slug, ...treatmentBranches.map((b) => b?.slug)].forEach(
+    (branch) => {
+      if (branch) {
+        slugs.push({ value: branch, label: branch });
+      }
+    },
+  );
+  return (
+    <>
+      <h3 className="h5 mb-3" id="previewURL">
+        Preview URL
+      </h3>
+      <p>
+        <b>Click to copy the link</b> you want to preview and paste it in your
+        browser. You need to enable <code>nimbus.debug</code> pref before.
+      </p>
+      <form data-testid="preview-url-form">
+        <label htmlFor="preview-branch">Selected Branch</label>
+        <Select
+          onChange={(e) => setSelection(e ? e.value : referenceBranch.slug)}
+          defaultValue={slugs[0]}
+          options={slugs}
+          name="preview-branch"
+          inputId="preview-branch"
+        />
+      </form>
```

Comment:
Nit: this could use a bottom margin

![Screen Shot 2021-05-10 at 9 30 00 AM](https://user-images.githubusercontent.com/6392049/117666972-67c0aa80-b172-11eb-96fd-89e64883b056.png)